src/Browser.php

```diff
@@ -784,7 +880,55 @@ private function requestMayBeStreaming($method, $url, array $headers = array(),
         }
 
         return $this->transaction->send(
-            new Request($method, $url, $headers, $body, $this->protocolVersion)
+            new Request($method, $url, $this->mergeHeaders($headers), $body, $this->protocolVersion)
         );
     }
+
+    /**
+     * @param array $headers
+     * @return array
+     */
+    private function mergeHeaders($headers)
+    {
+        foreach (array_keys($this->defaultHeaders) as $key) {
+            foreach (array_keys($headers) as $headerKey) {
+                if (strcasecmp($headerKey, $key) === 0) {
+                    unset($this->defaultHeaders[$key]);
```

Comment:
Unsetting the default headers seems like a fragile approach here? How about we iterative over the default headers and only *add* those that are not already present in `$headers` here?

Reply:
now addind it to the headers list instead of unsetting it from the default headers (no array_merge needed anymore)